src/components/SortByForm.jsx

```diff
@@ -1,42 +1,92 @@
+import { useNavigate } from "react-router-dom";
 
-export default function SortByForm ({ setOrderBy, setCategory, category }) {
-    const [ filterCategories, setFilterCategories ] = useState([])
-    const navigate = useNavigate()
+export default function SortByForm({
+  setOrderBy,
+  orderBy,
+  setCategory,
+  category,
+  setOrderDirection,
+  orderDirection,
+}) {
+  const [filterCategories, setFilterCategories] = useState([]);
+  const [ selectedOrder , setSelectedOrder ] = useState("")
+  const [ selectedDirection, setSelectedDirection ] = useState("")
 
-    useEffect(() => {
-        getCategories()
-        .then(({ data }) => {
-            setFilterCategories(data.categories)
-        })
-    }, [])
+  const navigate = useNavigate();
 
-    const handleFilterClick = (e) => {
-        e.preventDefault()
-        navigate(category)
-    }
+  useEffect(() => {
+    getCategories().then(({ data }) => {
+      setFilterCategories(data.categories);
+    });
+  }, []);
 
-    return (
-        <form className="sort-by-form">
-            <label htmlFor="category-filter">Filter by Category</label>
-            <select onClick={e => setCategory(e.target.value)} name="category" id="category-filter">
-                <option value="" selected>all</option> 
-                {filterCategories.map((category, i) => {
-                    const categoryName = category.slug.replaceAll("-", " ")
-                    return <option key={i} value={categoryName}>{categoryName}</option>
-                })}   
-            </select> 
-            <button onClick={handleFilterClick}>Filter</button><br />
-            <label htmlFor="sort-options">Order By</label>
-            <select name="sort" id="sort-options">
-                <option value="created_at">date</option>
-                <option value="title">title</option>
-                <option value="owner">review author</option>
-                <option value="votes">number of votes</option>
-            </select>
-            <button>Descending</button>
-        </form>
-    )
-}
\ No newline at end of file
+  const handleFilterClick = (e) => {
+    e.preventDefault();
+    navigate(category);
+  };
+
+  const handleOrderClick = (e) => {
+    e.preventDefault();
+    setOrderBy(selectedOrder)
+    setOrderDirection(selectedDirection)
+  };
+
+  return (
+    <form className="sort-by-form">
+      <label htmlFor="category-filter">Filter by Category</label>
+      <select
+        onClick={(e) => setCategory(e.target.value)}
+        name="category"
+        id="category-filter"
+      >
+        <option defaultValue="">
+          all
+        </option>
+        {filterCategories.map((category, i) => {
+          const categoryName = category.slug.replaceAll("-", " ");
+          return (
+            <option key={i} value={categoryName}>
+              {categoryName}
+            </option>
+          );
+        })}
+      </select>
+      <button onClick={handleFilterClick}>Filter</button>
```

Comment:
You can set `type="button"` instead of preventing the default behaviour here. A button's default type is `submit` which is why it'll submit the form unless it's changed or prevented.